simul/platform/platform.go

```diff
@@ -49,14 +51,28 @@ type Platform interface {
 }
 
 var localhost = "localhost"
+var amazonAWS = "aws"
+
+//var regions = []string{"us-west-2"}
 
 // NewPlatform returns the appropriate platform [deterlab,localhost]
 // and setups the Cleanup call in case of a signal interruption
-func NewPlatform(t string) Platform {
+func NewPlatform(t string, parameters map[string]string) Platform {
 	var p Platform
 	switch t {
 	case localhost:
 		p = NewLocalhost()
+	case amazonAWS:
+		pemFile, ok := parameters["pemFile"]
+		if !ok {
+			panic("pemFile file is missing")
+		}
+		regions := parameters["regions"]
+		regArray := strings.Split(regions, " ")
```

Comment:
MMhhhh have you tested with different regions like this already ? `reg1 reg2 reg3` -> I'm fearing that `reg2` and `reg3` will just be considered as regular command argument and not linked to the `regions` flag.

Overall, it may be worth it *later* to put all information needed in a config file and run the simul with `-aws config.toml` ..

Reply:
It has to be run with -regions "reg1 reg2 reg3" (added explanation in flag description), don't like it but I can't fin another way.Yeh ideally platform specific settings should be in the dedicated config file. Leaving it for now but also created separated issue for this